pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -118,7 +121,10 @@ func NewEndpointController(podInformer coreinformers.PodInformer, serviceInforme
 	e.eventBroadcaster = broadcaster
 	e.eventRecorder = recorder
 
-	e.endpointUpdatesBatchPeriod = endpointUpdatesBatchPeriod
+	// TODO(mborsz): Make burst adjustable.
+	// rate.Every(0) == rate.Inf.
+	e.syncRateLimiter = workqueue.NewItemBucketRateLimiter(rate.Every(endpointUpdatesBatchPeriod), 1)
```

Comment:
The goal is that when this changes this rapidly in succession, the first B (== 1) changes get quick service but then it gets limited? Can you please state this in the comment? It's not at all obvious.

Reply:
I don't see why the usual exponential backoff wouldn't work, there should be settings that approximate a step function like this?

Reply:
Also we usually rate limit on errors (?), but you're using it for ordinary processing, it might be good to state that.

Reply:
> The goal is that when this changes this rapidly in succession, the first B (== 1) changes get quick service but then it gets limited? Can you please state this in the comment? It's not at all obvious.

Yes. And +1 for making it more explicit.

> I don't see why the usual exponential backoff wouldn't work, there should be settings that approximate a step function like this?

Exponential is also possible, though because we're not using it for errors, it may be too strict.

Reply:
Added a comment.